squire_lib/src/ffi_tournament.rs

```diff
@@ -53,6 +55,36 @@ impl TournamentId {
         return ptr;
     }
 
+    /// Adds a player to a tournament
+    /// On error a NULL UUID is returned
+    #[no_mangle]
+    pub extern "C" fn tid_add_player(self: Self, __name: *const c_char) -> PlayerId {
```

Comment:
Is there a reason that these methods take `Self` and not `&Self`?

Reply:
rust didn't complain about no & is the reason

Reply:
Ah, gotcha. That makes sense I guess. `Uuid` (and by extension `TypeId`) implements `Copy`. You should be able to get rid of `: Self` and have the method take `self` (or `&self`).